pkg/deployer/helm/controller.go

```diff
@@ -130,11 +131,23 @@ func (a *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
-func (a *controller) reconcile(ctx context.Context, deployItem *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) error {
+func (a *controller) reconcile(ctx context.Context, deployItem *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) (err error) {
 	if len(deployItem.Status.Phase) == 0 {
 		deployItem.Status.Phase = lsv1alpha1.ExecutionPhaseInit
 	}
 
+	// set failed state if the last error lasts for more than 5 minutes
+	defer func() {
+		// set the error if the err is a landscaper error
+		if lsErr, ok := lsv1alpha1helper.IsError(err); ok {
+			deployItem.Status.LastError = lsErr.UpdatedError(deployItem.Status.LastError)
+		}
+		deployItem.Status.Phase = lsv1alpha1.ExecutionPhase(lsv1alpha1helper.GetPhaseForLastError(
+			lsv1alpha1.ComponentInstallationPhase(deployItem.Status.Phase),
+			deployItem.Status.LastError,
+			5*time.Minute))
+	}()
+
```

Comment:
```suggestion
```

Not needed as this behavior is already done in the calling function: https://github.com/gardener/landscaper/blob/master/pkg/deployer/helm/controller.go#L116

Reply:
Removed.